[DOCS-1166][DOCS-880][DOCS-1167][DOCS-1168] Update automation docs (#1084)
wandb/docs: content/guides/models/automations/_index.md

Reviewer: A few high level thoughts:
- For any numbered workflows describing steps in the UI I think it is valuable to have a gif that corresponds to whatever steps we are trying to show 
- The main thing I was hoping to address in this rewrite based on user feedback we got was to reduce the length of the one landing page. It's extremely long and overwhelming to find what you need to find. I think we should structure the automations section into several sub-sections, each a job to be done or a concrete workflow a user might come in trying to understand how to do it from start to finish. Right now it looks like everything is piled into _index which is super long and the user doesn't know what actions (i.e "how to troubleshoot your webhook endpoint") they can find in this doc.

Author: 1. Let's hold off on generating gifs yet. A gif, especially an animated one, suffers from technical debt each time we update the UI, even a minor update,. Also, images are large compared to paragraphs and can make the words seem less legible / can increase the cognitive load of reading the page. First I'd like to make sure that the words are as understandable as possible.
2. Done! We now have a landing page, an "Event scopes" reference page, a "Create automations" landing page, and specific pages each for Slack and webhook. I left the example payloads and troubleshooting details in the webhook creation page to try to keep the whole task together. I also removed the previous model-registry-automations and project-scoped-automations pages again, which crept back into this PR somehow.

PTAL!

  default:
    identifier: automations
    parent: w-b-models

> title: Create and manage automations

Reviewer 2: I am totally in favor of action-based titles. Just a heads up, we use the product name because Lukas (CEO) raged quit my action-based titles (years ago). That's to say, you might get feedback to change this to "Automations"

Author: Reverted this change for now

weight: 4
---

This page describes _automations_ in W&B and shows how to create and manage them. Create an automation to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) or a [registered model is changed]({{< relref "/guides/models/registry/" >}}).

An automation defines the [event scopes and types]({{< relref "#event-scopes-and-types" >}}) to watch for and the [action]({{< relref "#actions" >}}) to take when the event occurs, such as running a webhook or posting to a Slack channel.

## Events and event scopes

An automation can run when a specific event occurs at a given scope, either a registry or a project.

### Registry

> For an Artifact in a [Registry]({{< relref "/guides/models/registry/">}}) or a collection, [registered model]({{< relref "/guides/models/registry/">}}), you can configure an automation to run on these events:

Reviewer: ```suggestion
For an artifact in [Registry]({{< relref "/guides/models/registry/">}}), you can configure an automation to run on these events:
```

Author: Done

- **Linking a new Artifact to a collection**: Test and validate new models, datasets, etc.

> - **Adding a new alias to a version of an Artifact**: Trigger a special step of your workflow when a new Artifact version has a specific label or alias applied. For example, deploy a model when it has the `deploy` alias applied.

Reviewer: ```suggestion
- **Adding a new alias to a version of an Artifact**: Trigger a special step of your workflow when a new Artifact version has a specific alias applied. For example, deploy a model when it has the `deploy` alias applied.
```
there is no such thing as a label - you can only trigger an automation from adding an alias to an artifact version

Author: Done

### Project

> For an Artifact in a project or a collection, you can configure an automation to run on these events:

Reviewer: ```suggestion
For an artifact in a project, you can configure an automation to run on these events:
```
I think any references to an artifact should be lowercase, reference to the product, Artifacts should be uppercase.

Author: Done

- **Linking a new Artifact**: Test and validate new models, datasets, etc.
- **Creating a new version of an Artifact**: Apply recurring actions to each version of an Artifact. For example, start a training job when a new dataset artifact version is created.
- **Adding a new alias to a version of an Artifact**: Trigger a special step of your workflow when a new Artifact version in a project or collection has a specific label or alias applied. For example, , run a series of downstream processing steps when an Artifact has the `test-set-quality-check` alias applied.

> ## Event actions {#actions}

Reviewer: I'm a bit confused by this section name - event = if this, action = then that - what is an event action?

Author: Updated to just Actions

> When the configured event occurs, an automation runs the specified action, either by posting to a Slack channel or by running a webhook.

Reviewer: > either by posting to a Slack channel
I think this is only related to run metrics - is posting to a slack channel a new kind of action that is available now?

Author: Yes, there is a new way to integrate with Slack using Slack's oauth flow, rather than using Slack via a webhook. It was enabled only for run metrics for a while but now it is available for all automations.

1. To integrate with a new Slack workspace and channel, click **Connect Slack**.
    {{% alert %}}
    To integrate with a new Slack channel in a workspace that is already set up, click **New integration**.
    {{% /alert %}}

  If necessary, sign in to your Slack instance. A browser window appears, asking that you grant Weights and Biases permission to post to the Slack channel you select. Read the page, then click **Search for a channel** and begin typing the channel name. Select the channel from the list, then click **Allow**.

1. In Slack, go to the channel you selected. If you see a post like `[Your Slack handle] added an integration to this channel: Weights & Biases`, the integration is configured correctly.

Now you can [create an automation that uses the Slack integration]({{< relref "#create-slack-automation" >}}).

#### Create a Slack automation {#create-slack-automation}

After you [configure a Slack integration]({{< relref "#add-a-slack-integration" >}}), follow these steps to create a Slack automation that uses it.

1. Log in to W&B and go to the project page.
1. In the sidebar, click **Automations**.
1. Click **Create automation**.

> 1. Choose the [**Event**]({{< relref "#events-and-event-scopes" >}}) which triggers the automation. If applicable, provide options that are specific to the event type. If your project has no registries, registry events will not be available. Click **Next step**.

Reviewer: ```suggestion
1. Choose the [**Event**]({{< relref "#events-and-event-scopes" >}}) which triggers the automation. If applicable, provide options that are specific to the event type. If your project has no registries, registry events will not be available. Click **Next step**.
```
What do you mean by "If your project has no registries, registry events will not be available." ? Is this trying to say that if you're created an automation inside a project, you will not see any registry related events as you're in the project scope, not the registry scope? 

A project does not contain registries - inside an org, you have some projects, and some registries, but registries are not nested inside of projects. They're at the same "level". 

The more general comment it sounds like you're trying to call out is that you will only see events available that are relevant o the scope you are in. So if you are created an automation inside a project, you will only see project events, if you are creating an automation inside a registry, you will only see registry events.

Author: Yes that's what I mean. Let me try again.

1. Select the team where you added the Slack integration.
1. Set **Action type** to **Slack notification**.
1. Select the Slack channel, then click **Next step**.
1. Provide a name for the automation. Optionally, provide a description.
1. Click **Create automation**.

> ### Configure a webhook automation

Reviewer: Here is an example of a task that should be a separate sub-page so a user can easily access it from the sidebar.

Author: I'm not sure about this, because this page is about creating automations. I could imagine creating the webhook itself being on a separate page because the automation depends on it, but I don't think a webhook is good for anything other than an automation, right? There's a benefit in keeping it all on one page. What do you think?

Author: Done

1. Go to the **Settings** page.
1. In the **Webhooks** section, click **New webhook**.
1. Provide a name for the webhook. 
1. Provide the endpoint URL for the webhook.
1. If the webhook requires an access token or any other [secrets]({{< relref "/guides/core/secrets.md" >}}), grant the webhook access to each secret by setting **Secret** to the secret's name. When you configure the automation that uses the webhook, you can access the secret in the payload by prefixing its name with `$`.
1. If the webhook authenticates using an access token, set **Access token** to the name of the secret that contains it. When you configure an automation that uses the webhook, you can access the token in the `$ACCESS_TOKEN` environment variable, and the HTTP header sets `Authorization: Bearer` to the access token.
1. Click **Test** to test the webhook. Optionally, provide a payload to test. To refer to a secret the webhook has access to in the payload, prefix its name with `$`. Any payload you specify in this step does not persist, but allows you to test authentication. You configure an automation's payload when you [create the automation]({{< relref "#create-webhook-automation" >}}).

{{% alert %}}

> See [Troubleshoot your webhook]({{< relref "#troubleshoot-your-webhook" >}}) to view where the secret and access token are specified in

Reviewer: I think users would find it helpful to see the webhook structure actually in this section as well, as they are in the process of constructing it. Again, I think have a whole section just on creating a webhook, just as you had done in here for secrets would be beneficial. 

https://www.notion.so/wandbai/W-B-Post-Request-Structure-1a7e2f5c7ef380a6a446fce56201b00c?pvs=4 - we can include this structure information as well to help users as they are creating it (that doc is just an idea for how to explain the structure, we don't have to use that text exactly).

Reviewer: Here is how product board shows the request structure in their developer docs - https://developer.productboard.com/reference/postwebhook-1

Author: I think this is more relevant to the automation API docs than for the _guide_ about creating webhooks. The new webhook page includes the examples and the troubleshooting content. The API updates are a separate effort that is not in scope for this PR.

Author: Captured to https://wandb.atlassian.net/browse/DOCS-1379

the POST request.
{{% /alert %}}

Now you can [create a webhook automation]({{< relref "#create-webhook-automation" >}}).

#### Create the automation {#create-webhook-automation}

> 1. Log in to W&B and go to the project page.

Reviewer: Hmm so here is where the nuance of the scope matters. The most common workflow is for users to navigate to a registry and create an automation in there. The instruction defined above is only relevant if you want to create an automation inside a project. And in registry, an automation can either be at the collection or registry level (i.e applies to all collection in a registry) - I think we would want a separate sub-page on the scopes available and how to create an automation for each.

Author: PTAL, I have now structured the Project and Registry tasks into tabs, filled in missing details about both registry collections and project collections, and expanded the Events and Scopes page.

1. In the sidebar, click **Automations**.
1. Click **Create automation**.
1. Choose the **Event** which triggers the automation. If applicable, provide options that are specific to the event type. If your project has no registries, registry events will not be available. Click **Next step**.
1. Select the team where you added the webhook.
1. Set **Action type** to **Webhook**, then select the webhook.
1. Provide the payload for the webhook in **Payload**. Refer to [Example webhook payloads]({{< relref "#example-webhook-payloads" >}}) for variables you can use. If the [webhook]({{< relref "#add-a-webhook" >}}) has access to any secrets, you can access the secret in the payload by prefixing its name with `$`. If the webhook has an access token, you can access the token in the `$ACCESS_TOKEN` environment variable, and the HTTP header sets `Authorization: Bearer` to the access token.
1. Click **Next step**.
1. Provide a name for the automation. Optionally, provide a description.
1. Click **Create automation**.

> ### Example webhook payloads

Reviewer: I think this should also be its own separate section

Author: I'm not sure how many separate pages we want the reader to have to read to do the task. I'm trying to keep the content all together since they have to do it all together. WDYT?

Author: Restructuring took care of this, I think.

> The following tabs demonstrate example payloads based on common use cases. Within the examples they reference the following keys to refer to condition objects in the payload parameters:

Reviewer: There is a also a metadata template string that is not in this list that we should add

Author: Thanks. Where can I learn more about it?

The following is an example payload:

```json
{
    "text": "New alert from WANDB!",
"blocks": [
    {
            "type": "section",
        "text": {
            "type": "mrkdwn",
            "text": "Registry event: ${event_type}"
        }
    },
        {
            "type":"section",
            "text": {
            "type": "mrkdwn",
            "text": "New version: ${artifact_version_string}"
        }
        },
        {
        "type": "divider"
    },
        {
            "type": "section",
        "text": {
            "type": "mrkdwn",
            "text": "Author: ${event_author}"
        }
        }
    ]
}
```

{{% /tab %}}
{{< /tabpane >}}

### Troubleshoot your webhook

Interactively troubleshoot your webhook with the W&B App UI or programmatically with a Bash script. You can troubleshoot a webhook when you create a new webhook or edit an existing webhook.

{{< tabpane text=true >}}
{{% tab header="W&B App UI" value="app" %}}

> Interactively test a webhook with the W&B App UI. 

Reviewer: lets be more specific on what exactly we are letting the user test (webhook is a bit too vague) - here we are letting them test whether the webhook "address" is correct and whether the receiving endpoint can receive the webhook W&B would trigger on the users behalf.

Author: Good point, done!

  default:
    identifier: automations
    parent: w-b-models
title: Create and manage automations
weight: 4
---

This page describes _automations_ in W&B and shows how to create and manage them. Create an automation to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) or a [registered model is changed]({{< relref "/guides/models/registry/" >}}), or based on the results of a [run]({{< relref "/guides/models/track/runs/manage-runs.md" >}})

An automation defines the [event scopes and types]({{< relref "#event-scopes-and-types" >}}) to watch for and the [action]({{< relref "#event-actions" >}}) to take when the event occurs, such as running a webhook or posting to a Slack channel.

## Event scopes and types

An automation's triggering event depends on the automation's scope.

### Registry

> For a [registered model]({{< relref "/guides/models/registry/">}}), you can configure an automation to run on these events:

Reviewer 3: I believe there's no restriction to just the registered model types, it's any datatype in any collection.

I'm pretty sure the term "registered model" is not a thing anymore, it's an artifact (!) of when that was the only type of registry, but I might be wrong.
(more instances of this below, but I won't call them out anymore)

> - **Linking a new artifact to a registered model**: Test and validate new model candidates.

Reviewer 4: **suggestion:**  Since we're trying to move away from language that restricts artifacts to models alone:
```suggestion
- **Linking a new artifact to a collection**: Test and validate new models, datasets, etc.
```

Reviewer 4: **note**: Also, this event is supported at the project scope

Author: Done, PTAL

> - **Adding a new alias to a version of the registered model**: Trigger a special step of your workflow when a new model version has a label or alias applied. For example, deploy a model when it has the `deploy` alias applied.

Reviewer 4: **note:** This event is supported at both the "project" and "artifact collection" scope

Author: Done, PTAL

### Project

> For a model in a project that is not in a registry, you can configure an automation to run on these events:

Reviewer 3: ditto above - not just about models, any artifact.

- **A new version of an artifact is created in a collection**: Apply recurring actions to each version of an artifact. For example, start a training job when a new dataset artifact version is created.

> - **An artifact alias is added**: Apply recurring actions when a specific alias is applied to an artifact version. For example, run a series of downstream processing steps when an artifact gains the `test-set-quality-check` alias.

Reviewer 4: **note:** These events can be scoped at either the `project` _or_ `artifact collection` level as well

Author: Done

> ### Run

Reviewer 3: (I know we discussed in slack, but flagging here since it's convenient) We should keep mentions of the run metric automations out of the docs until the feature is generally released.

You can configure an automation to run based on the results of a run:

- **Run metric threshold met**: Apply recurring actions based on a run metric's value and whether it is above, below, or exactly equal to the threshold. Optionally, specify how to average results.

> - **Run metric change threshold met**: Apply recurring actions based on whether a run metric's amount of change (absolute or relative) is above, below, or exactly equal to the threshold. Optionally, specify how to average results.

Reviewer 3: the options for change thresholds are different enough that the descrption should be different - eg the amount of change is no above/below/exactly equal to

Author: I'll implement this in the followup PR.

> - **Run metric change threshold met**: Apply recurring actions based on whether a run metric's amount of change (absolute or relative) is above, below, or exactly equal to the threshold. Optionally, specify how to average results.

Reviewer 4: **note:** While I can see why it'd be helpful to separate this section out, run metric events are all project-scoped, so this should probably be a subsection inside **Project**

> ## Evant actions

Reviewer 3: ```suggestion
## Event actions
```

Reviewer 3: I think it's probably worth keeping the concepts for events and actions separate, so I think it's a bit confusing to call something "event actions" :) 

I would probably just call them actions

An automation can run a webhook on a third-party service such as GitHub or Microsoft Teams, or it can post to a Slack channel.

## Create an automation
These sections show how to configure a Slack automation or a webhook automation.

### Configure a Slack automation
Configuring a Slack integration takes multiple steps. First, [add one or more Slack channels]({{< relref "#add-a-slack-channel" >}}) as alert destinations. Next, [create the automation that notifies the Slack channel]({{< relref "#create-slack-automation" >}}).

#### Add a Slack channel

1. Go to Team Settings page.
1. To integrate with a new Slack workspace and channel, click **Connect Slack**.
    {{% alert %}}
    To integrate with a new Slack channel in a workspace that is already set up, click **New integration**.
    {{% /alert %}}

  If necessary, sign in to your Slack instance. A browser window appears, asking that you grant Weights and Biases permission to post to the Slack channel you select. Read the page, then click **Search for a channel** and begin typing the channel name. Select the channel from the list, then click **Allow**.

> 1. In Slack, go to the channel you selected. If you see a post like `[Your Slack handle] added an integration to this channel: Weights & Biases`, the integration is configured correctly.

Reviewer 3: @tonyyli-wandb is there any action that admins need to take to approve the integration?

Author: In my testing, I did not need anyone to approve it and I am not a Slack admin. I think Slack just posts using the permissions of the Slack user enabling the integration.

Now you can [create an automation that uses the Slack integration]({{< relref "#create-slack-automation" >}}).

#### Create the automation {#create-slack-automation}

After you [configure a Slack integration]({{< relref "#configure-the-slack-integration" >}}), follow these steps to create an automation that uses it.

1. Go to the project page.
1. In the sidebar, click **Automations**.
1. Click **Create automation**.
1. Choose the **Event** which triggers the automation. If applicable, provide options that are specific to the event type. If your project has no registries, registry events will not be available. Click **Next step**.
1. Select the team where you added the Slack integration.
1. Set **Action type** to **Slack notification**. Select the Slack channel, then click **Next step**.

> 1. Proviude a name for the automation. Optionally, provide a description.

Reviewer 3: ```suggestion
1. Provide a name for the automation. Optionally, provide a description.
```

1. Click **Create automation**.

### Configure a webhook automation

Configuring a webhook integration takes multiple steps:

> 1. If your webhook requires any sensitive strings, [add them as secrets]({{< relref "#add-a-secret" >}}) for any sensitive strings required by the webhook. If the webhook requires a bearer token, do not configure a secret for it. Instead, you configure it as part of creating the webhook.

Reviewer 3: you repeat the "any sensitive strings" phrase

Reviewer 3: Based on the current UI, I believe bearer tokens are also pulled from secrets

Author: Nope, it's a separate Access Token field. You paste it right into the team-level webhook where you provide the endpoint URL.

Reviewer 3: yes, it's a separate field, but the list you're selecting from (for both Secret and Access Token fields) is the list of secrets.

Note that when you open the dropdown for both access token and secret fields, it's the same list of stuff 
![image](https://github.com/user-attachments/assets/93318023-b6fb-4b68-807d-f4a2d64340c8)
![image](https://github.com/user-attachments/assets/50f49b8b-bb4e-4584-bee6-49e58c382dc2)

Author: Done!

1. [Create the webook]({{< relref "#add-a-webhook" >}}). If necessary, specify its access token, and grant it access to any secrets it needs.

> 1. [Create an automation that uses the webhook]({{< relref "#create-webhook-automation" >}}), configuring the payload it sends to the third-party service.

Reviewer 3: webhooks don't necessarily go to third party services, so may not be useful to say that here.

Author: Done

> 1. [Create an automation that uses the webhook]({{< relref "#create-webhook-automation" >}}), configuring the payload it sends to the third-party service.

Author: ```suggestion
1. [Create an automation that uses the webhook]({{< relref "#create-webhook-automation" >}}), configuring the payload it sends.
```

> #### Add a secret

Reviewer 3: should this be on its own page? AFAIK they are used elsewhere and are not automations specific, but I could be wrong.

Author: Surprisingly, we do not seem to have any existing docs about secrets. I'm not aware of where all you can use them. cc/ @tristan-spaulding2 who might know more?

Author: PTAL, I've now separated Secrets out into its own page. I'm not aware of anything else that can use a secret currently.

> A secret is a team-level variable that lets you obfuscate a sensitive string such as a credential, API key, password, or token. W&B recommends you use secrets to store any string that you want to protect the plain text content of.

Reviewer 3: I know you're just copying the earlier text, but I would definitely avoid using the term `obfuscate` here - in security settings, it implies that we are not storing the secret in a secure manner. 

an AI summary I found describes them as "a piece of sensitive information, like a password, API key, or certificate, that is securely stored and managed within the Secret Manager service, allowing applications to access this data without exposing it directly within the code"

emphasizing that they are securely stored and then that you can use them without exposing them everywhere (eg in the automation config) are the main points to hit. (we should verify that we actually secure store them :)

Author: Not yet done.

Author: Done, i think!

  - W&B does not recommend that you use a Kubernetes cluster as the backend of your secrets store. Consider a Kubernetes cluster only if you are not able to use a W&B instance of a cloud secrets manager (AWS, GCP, or Azure), and you understand how to prevent security vulnerabilities that can occur if you use a cluster.
{{% /alert %}}

To add a secret:

1. If necessary, generate the sensitive string in the webhook's service. For example, generate an API key or set a password. If necessary, save the sensitive string securely, such as in a password manager.
1. Log in to W&B and go to the **Settings** page.
1. In the **Team Secrets** section, click **New secret**.
1. Using letters, numbers, and `_`, provide a name for the secret.
1. Paste the sensitive string into the **Secret** field.
1. Click **Add secret**.

Specify the secrets you want to use for your webhook automation when you configure the webhook. See the [Configure a webhook]({{< relref "#configure-a-webhook" >}}) section for more information. 

{{% alert %}}

> Once you create a secret, you can access that secret in your W&B workflows with `$`.

Reviewer 3: I don't know what a `workflow` means here, and I don't really know what this mean. Does it mean that you can reference a secret in a webhook payload using $my_secret_name ?

Author: Yes, I believe so.

> Once you create a secret, you can access that secret in your W&B workflows with `$`.

Author: ```suggestion
Once you create a secret and grant the webhook access to it, you can use it in your automation's webhook payload by prefixing its name with with `$`.
```

> Once you create a secret and grant the webhook access to it, you can use it in your automation's webhook payload by prefixing its name with with `$`.

Author: ```suggestion
Once you create a secret and grant the webhook access to it, you can use it in your automation's webhook payload by prefixing its name with `$`.
```

{{% /alert %}}

#### Add a webhook

{{% alert %}}
* Only W&B Admins can configure a webhook for a W&B Team.
* Before you create a webhook, [create any secrets it needs]({{< relref "#add-a-secret" >}}).
{{% /alert %}}

This section shows how to configure a webhook's URL, as well as any access tokens and secrets it requires.

1. If the webhook requires any access tokens or other sensitive strings, create a secret in W&B for each sensitive string, and make a note of the secret's name.
1. Log in to W&B and go to the **Settings** page.
1. In the **Webhooks** section, click **New webhook**.
1. Provide a name for the webhook. 
1. Provide the endpoint URL for the webhook.
1. If the webhook authenticates using an access token, set **Access token** to the name of the secret that contains the access token. When you configure an automation that uses this webhook, the access token will be available in the `$ACCESS_TOKEN` environment variable, and the HTTP header will have `Authorization: Bearer $ACCESS_TOKEN`.

> 1. If the webhook validates its payload using a sensitive string, set **Secret** to the name of the secret that contains the sensitive string.

Reviewer 3: So this doesn't really tell the user what happens to this value when we send the webhook. Is it sent in the body of the payload? is it available in the payload as $my_secret_name? is it sent in a header named secret? 

So, TODO item for me: figure out what this does (unless @tonyyli-wandb knows off the top of your head?)

Author: Let me know what you find out!

Author: I believe I've addressed this now.

1. Click **Test** to test the webhook. Optionally, provide a payload to test. Any payload you specify in this step does not persist, and will need to be specified when you [create the webhook automation]({{< relref "#create-webhook-automation" >}}).

{{% alert %}}

> See the [Troubleshoot your webhook]({{< relref "#troubleshoot-your-webhook" >}}) section to view where the secret and access token are specified in

Reviewer 3: my opinion: we should just remove this alert and fix the above to tell the user :)

Author: I think I took care of this!

the POST request.
{{% /alert %}}

Now you can [create an automation that uses the webhook]({{< relref "#create-webhook-automation" >}}).

#### Create the automation {#create-webhook-automation}
1. Log in to W&B and go to the project page.
1. In the sidebar, click **Automations**.
1. Click **Create automation**.
1. Choose the **Event** which triggers the automation. If applicable, provide options that are specific to the event type. If your project has no registries, registry events will not be available. Click **Next step**.
1. Select the team where you added the webhook.
1. Set **Action type** to **Webhook**, then select the webhook.

> 1. Provide the payload for the webhook in **Payload**. Refer to the reference for varialbles that you can use. For details, refer to the [Example webhook payloads]({{< relref "#example-webhook-payloads" >}}) section.

Reviewer 3: ```suggestion
1. Provide the payload for the webhook in **Payload**. Refer to the reference for variables that you can use. For details, refer to the [Example webhook payloads]({{< relref "#example-webhook-payloads" >}}) section.
```

1. Click **Next step**.
1. Provide a name for the automation. Optionally, provide a description.
1. Click **Create automation**.

### Example webhook payloads

The following tabs demonstrate example payloads based on common use cases. Within the examples they reference the following keys to refer to condition objects in the payload parameters:
* `${event_type}` Refers to the type of event that triggered the action.
* `${event_author}` Refers to the user that triggered the action.

> * `${artifact_version}` Refers to the specific artifact version that triggered the action. Passed as an artifact instance.

Reviewer 3: We need to answer the question: what does it mean that something is 'passed as an artifact instance'?

Author: Good idea. This content is copied from where it lives in the current doc, not net new in this PR.

Author: https://docs.wandb.ai/ref/python/artifact/
